added parquet-encoder and test files

4hlberg opened the pull request:
Not all functionality is in here yet. Read and write based on schema works.

* Should add parser from json to parquet-schema string for cleanliness
* Should add functionality to handle deleted and recoded timestamps in the mapper.

Also:
Look at objectstorage to make sure all functionality is there.

4hlberg committed:
added parquet-encoder and test files

gra-moore commented on encoder/parquet.go:
```diff
@@ -0,0 +1,289 @@
+		return nil, err
+	}
+	var flushThreshold int64
+	if sourceConfig["flush_threshold"] == 0 {
```
This can be done after doing the parsing. That would be cleaner as works with the struct and no need for accessing the map.

gra-moore commented on encoder/parquet.go:
```diff
@@ -0,0 +1,289 @@
+	// enc/dec should not have a concept of flush?
+	written := c.writer.CurrentRowGroupSize()
+	// log something here?
+	fmt.Sprintf("Written %v parquet bytes to underlying writer. written in total: %v", written, c.writer.CurrentFileSize())
```
remove print

gra-moore commented on encoder/parquet.go:
```diff
@@ -0,0 +1,289 @@
+	}
+	flushed := written - c.writer.CurrentRowGroupSize()
+	// log something? is it really needed?
+	fmt.Sprintf("Flushed %v parquet bytes to underlying writer. flushed in total: %v", flushed, c.writer.CurrentFileSize())
```
remove print

gra-moore commented on encoder/parquet_test.go:
```diff
@@ -0,0 +1,163 @@
+	if item == nil {
+		t.Error("Expected item")
+	}
+	fmt.Sprint(item.GetValue("name"))
```
remove

gra-moore reviewed (changes requested):
Looking good. A couple of comments.

4hlberg committed:
Removed debugging-print
Move some checks to flushThreshold, remove logging until logger can be
passed down with the interfaces created.

4hlberg committed:
Merge branch 'master' into feature/parquet-encoder

4hlberg committed:
Added readme example for the parquet-encoder
